### ISSUE TITLE
Is there any special reason to implement PL loss in a sligntly different way from paper?

### ISSUE THREAD

**Issue by SongweiGe** [lucidrains/stylegan2-pytorch#78]
Hey thanks for releasing the code! It worked pretty good for the most of time except that sometimes the PL loss exploded in the first few steps. Then I found that the PL loss is implemented in a slightly different way from directly calculating the gradient of g(w)*y: the code estimates the gradient of generated image with respect to w by perturbating w a little bit. I think that's reasonable, but is there any other special reason to use that? Could that be the reason that PL loss exploded during the beginning of the training? I really appreciate your anwsers!

**Comment by lucidrains** [comment]
Hi @SongweiGe ! I actually originally transcribed this work from a tensorflow 2 implementation, and that was the way they had it. I have rewritten it to the way it was done officially https://github.com/lucidrains/stylegan2-pytorch/commit/a88681ee514b9b59b933b8d24f56f8cf0253d5aa Hope this adds some stability and thanks for bringing it to my attention

**Comment by SongweiGe** [comment]
Cool! Thanks for the very quick reply and help!

**Comment by SongweiGe** [comment]
Hey @lucidrains, a quick question, is gradient_accumulate_every=5 and batch_size=6 the same with gradient_accumulate_every=1 and batch_size=30?

**Comment by lucidrains** [comment]
@SongweiGe yes!

**Comment by lucidrains** [comment]
@SongweiGe what are you training on, and how does the results look?

**Comment by SongweiGe** [comment]
I've been working on generating sketches with GANs. So far I have it worked on some simple datasets but it fails when the sketches become more complex. I'm going to add more data augmentation tricks to see if that would help. For now, I only have around 10k data.
